src/main/java/budgetbuddy/model/ModelManager.java

```diff
@@ -28,20 +29,22 @@
     /**
      * Initializes a ModelManager with the given addressBook and userPrefs.
      */
-    public ModelManager(LoansManager loansManager, ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs) {
+    public ModelManager(LoansManager loansManager, AccountsManager accountsManager, ReadOnlyAddressBook addressBook,
+                        ReadOnlyUserPrefs userPrefs) {
         super();
         requireAllNonNull(loansManager, addressBook, userPrefs);
 
         logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs);
 
         this.loansManager = new LoansManager(loansManager.getPersonsList());
+        this.accountsManager = new AccountsManager(accountsManager.getAccountsList());
```

Comment:
This is really strange. Why create a new AccountsManager when one is passed in?

Reply:
yeah... but the other variables are also created this way, so i just followed it for consistency